File: January 2023/24 January 2023/Convert an array to reduced form.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>

using namespace std;


// } Driver Code Ends
//User function template for C++
class Solution{
public:	
	void convert(int arr[], int n) {
	    
	    vector<pair<int, int>> ans;
	    
	    for (int i=0; i<n; i++){
	        ans.push_back({arr[i], i});
	    }
	    
	    sort(ans.begin(), ans.end());
	    
	    for (int i=0; i<n; i++){
	        arr[ans[i].second] = i;
	    }
	}

};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        int arr[n];
        for (int i = 0; i < n; i++) {
            cin >> arr[i];
        }
        Solution ob;
        ob.convert(arr, n);
        for (int i = 0; i < n; i++) {
            cout << arr[i] << " ";
        }
        cout << "\n";
    }
    return 0;
}

// } Driver Code Ends

File: January 2023/24 January 2023/Convert an array to reduced form.py
#User function Template for python3
class Solution:

	def convert(self,arr, n):
		brr = sorted(arr)
        lst = {}
        for i in range(n):
            lst[brr[i]]=i
        for j in range(n):
            arr[j]=lst[arr[j]]


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    tc = int(input())
    while tc > 0:
        n = int(input())
        arr = list(map(int, input().strip().split()))
        ob = Solution()
        ob.convert(arr, n)
        for x in arr:
            print(x, end=" ")
        print()
        tc -= 1

# } Driver Code Ends
